### PR TITLE
[Enhancement] Limit err msg size if there is an exception when converting JSON when loading

### PR DESCRIPTION
**Pull request by srlch**
## Why I'm doing:
If strip_outer_array is true, we may get huge of json err msg if the converting got an exception. It is because simdjson's api can only get raw json string of the entire json array.

## What I'm doing:
Limit the err msg when JSON converting  got an exception.

Fixes #issue
https://github.com/StarRocks/StarRocksTest/issues/7978

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by wyb — DISMISSED** [review]

File: be/src/util/json_converter.cpp [hunk 1]
```diff
@@ -37,8 +38,10 @@ class SimdJsonConverter {
             return JsonValue(builder.slice());
         } catch (simdjson::simdjson_error& e) {
             std::string_view view(value.get_raw_json_string().raw());
-            auto err_msg = strings::Substitute("Failed to convert simdjson value, json=$0, error=$1", view.data(),
-                                               simdjson::error_message(e.error()));
+            // truncate the raw json string if it is too large
+            size_t size = view.size() > MAX_VALUE_LENGTH_FOR_ERRMSG ? MAX_VALUE_LENGTH_FOR_ERRMSG : view.size();
```
Comment (kevincai) [inline be/src/util/json_converter.cpp:42]: usually append addition error messages if is truncated, e.g. `'original_json_string ... <truncated> `

File: be/src/util/json_converter.cpp [hunk 2]
```diff
@@ -37,8 +38,16 @@ class SimdJsonConverter {
             return JsonValue(builder.slice());
         } catch (simdjson::simdjson_error& e) {
             std::string_view view(value.get_raw_json_string().raw());
-            auto err_msg = strings::Substitute("Failed to convert simdjson value, json=$0, error=$1", view.data(),
-                                               simdjson::error_message(e.error()));
+            // truncate the raw json string if it is too large
```
Comment (kevincai) [inline be/src/util/json_converter.cpp:41]: it should be much easier to cover this by an unit test than a sql tester.

**Review by kevincai — DISMISSED** [review]

File: be/src/util/json_converter.cpp (hunk 2 above)
Comment (srlch) [reply be/src/util/json_converter.cpp:41]: updated

File: be/test/formats/json/nullable_column_test.cpp [hunk 3]
```diff
@@ -142,4 +143,12 @@ TEST_F(AddNullableColumnTest, test_add_map) {
 
     ASSERT_EQ("[{'key1':'foo','key2':'bar','key3':'baz'}]", column->debug_string());
 }
+
+TEST_F(AddNullableColumnTest, test_invalid_json_convert) {
+    simdjson::ondemand::parser parser;
+    auto json = R"(  { "key" : "abc \ssn\"" }  )"_padded;
+    auto doc = parser.iterate(json);
+    simdjson::ondemand::value val = doc.find_field("key");
+    (void)convert_from_simdjson(val);
```
Comment (kevincai) [inline be/test/formats/json/nullable_column_test.cpp:152]: it's a waste of the UT if bypass the return value validation. That's the whole point of this unit test. lol

Comment (kevincai) [reply be/test/formats/json/nullable_column_test.cpp:152]: you can even change the `MAX_VALUE_LENGTH_FOR_ERRMSG`, to validate that the json string in error message DOES get truncated as expected.

Comment (srlch) [reply be/test/formats/json/nullable_column_test.cpp]: updated

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by srlch** [comment]
https://github.com/Mergifyio backport branch-3.2

**Pull request merged**